packages/core/core/src/BundleGraph.js

```diff
@@ -1122,6 +1122,11 @@ export default class BundleGraph {
   }
 
   isAssetReferenced(bundle: Bundle, asset: Asset): boolean {
+    // If the asset is available in multiple bundles, it's referenced.
+    if (this.getBundlesWithAsset(asset).length > 1) {
```

Comment:
Just because an asset is in multiple bundles doesn't mean it is referenced _between_ multiple bundles though right? The references could be totally self contained within that bundle.

Reply:
Does Parcel really know how all assets are going to be loaded though? We have scenrios where we add bundles to HTML dynamically and Parcel isn't aware that the two are linked. I know this isn't ideal, but I feel like we should be resilient to these kinds of issues. 

Also check out the test I added, that was failing before I made this fix and that scenario should definitely work IMO.

Reply:
Right yeah I was more wondering about it because the of the name of the function like Niklas mentioned

Reply:
Yeah I hear that. I can move it the fix, however I feel like that leaves an issue in `isAssetReferenced`. Shouldn't it return true for the scenario I outlined in the test?

Reply:
depends on the definition of "referenced" I guess. If we mean "this exact instance of an asset is referenced from another bundle", then a duplicated asset wouldn't count. But if we want duplicated assets to behave at runtime like they aren't duplicated, then it would make sense.